File: day6/07-errorHandling/main.go
package main

import (
	"errors"
	"fmt"
	"math"
)

func sqrt(n int) (float64, error) {
	if n < 0 {
		return 0.0, errors.New("can't take square root of a negative number")
	}

	return math.Sqrt(float64(n)), nil
}

func main() {
	value, err := sqrt(-12)
	if err != nil {
		fmt.Println(err.Error())
		return
	}

	fmt.Println(value)
}

File: day6/07-errorHandling/Main.java
class Main {
  public static double sqrt(int n) throws Exception {
    if (n < 0) {
      throw new Exception("can't take square root of a negative number");
    }
    return Math.sqrt(n);
  }

  public static void main(String[] args) {
    try {
      double value = sqrt(-12);
    } catch (Exception err) {
      System.out.println(err.getMessage());
    }
  }
}
